src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -305,6 +331,37 @@ def start(
         for root_name, root_path in manifest_paths_by_root.items():
             session.add_manifest_path(root=root_name, path=root_path)
 
+        # Create WorkerManifestProperties list for enhanced worker agent processing
+        if not step_dependencies:
+            for manifest_properties in manifest_properties_list:
+                local_root_path: str = session._asset_sync.get_local_destination(
+                    manifest_properties=manifest_properties,
+                    dynamic_mapping_rules=dynamic_mapping_rules,
+                    storage_profiles_path_mapping_rules=storage_profiles_path_mapping_rules_dict,
+                )
+
+                # Create worker manifest property and add to session
+                worker_manifest_props = WorkerManifestProperties(
+                    manifest_properties=manifest_properties,
+                    local_root_path=local_root_path,
+                )
+                session.set_worker_manifest_properties(worker_manifest_props)
+
+        # Prepare input manifest for download task run
+        download_manifest_properties_list: list[WorkerManifestProperties] = list()
+        for local_root_path, manifest_path in manifest_paths_by_root.items():
+            session.add_local_manifest_path(
+                local_root_path=local_root_path, manifest_path=manifest_path
+            )
+            download_manifest = session.get_worker_manifest_properties(
+                local_root_path=local_root_path
+            )
+            assert download_manifest is not None
```

Comment:
Would `download_manifest` always be none if `step_dependencies` is falsy? or `session.set_worker_manifest_properties` being called on the session object outside of this file?